esmvalcore/preprocessor/_time.py

```diff
@@ -737,6 +737,46 @@ def regrid_time(cube, frequency):
     return cube
 
 
+def _get_time_bounds(time, freq):
+    bounds = []
+    for step, point in enumerate(time.points):
+        month = time.cell(step).point.month
+        year = time.cell(step).point.year
+        if freq in ['mon', 'mo']:
+            next_month = _get_next_month(month, year)
+            min_bound = time.units.date2num(
+                datetime.datetime(year, month, 1, 0, 0))
+            max_bound = time.units.date2num(
+                datetime.datetime(next_month[1], next_month[0], 1, 0, 0))
+        elif freq == 'yr':
+            min_bound = time.units.date2num(
+                datetime.datetime(year, 1, 1, 0, 0))
+            max_bound = time.units.date2num(
+                datetime.datetime(year + 1, 1, 1, 0, 0))
+        elif freq == 'dec':
+            min_bound = time.units.date2num(
+                datetime.datetime(year, 1, 1, 0, 0))
+            max_bound = time.units.date2num(
+                datetime.datetime(year + 10, 1, 1, 0, 0))
+        else:
+            delta = {
+                'day': 12 / 24,
```

Comment:
Are you talking about the cube in the tests? Because that cube has the midpoints at 00:00, so the bounds are at 12:00 because it's summing half a day. If the midpoints are at 12:00, I would say the bounds are going to be indeed at 00:00. The sum is the same half day.

Reply:
So the CMOR standards don't specify anything about the midpoints?

Reply:
Not that I know of (@jvegasbsc ?)

Reply:
There is no standard about the representative points, in fact they do not have to be midpoints: they can be at any point inside the time range you are covering. Remember that files  **should** have the bounds and the cell methods filled correctly so it is usually not a problem.

In general, I am quite sure about our fixes for monthly and lower frequencies, as I know better how the modellers usuallys set the representative points, but for daily and hourly data I am more uncertain